hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -857,21 +861,20 @@ public Response initializeMultipartUpload(
   }
 
   private ReplicationConfig getReplicationConfig(OzoneBucket ozoneBucket,
-      String storageType) throws OS3Exception {
-    if (StringUtils.isEmpty(storageType)) {
-      S3StorageType defaultStorageType = S3StorageType.getDefault(ozoneConfiguration);
-      storageType = (defaultStorageType != null ? defaultStorageType.toString() : null);
-    }
+      String storageType, String storageConfig) throws OS3Exception {
 
     ReplicationConfig clientConfiguredReplicationConfig = null;
     String replication = ozoneConfiguration.get(OZONE_REPLICATION);
+
     if (replication != null) {
-      clientConfiguredReplicationConfig = ReplicationConfig.parse(
-          ReplicationType.valueOf(ozoneConfiguration
-              .get(OZONE_REPLICATION_TYPE, OZONE_REPLICATION_TYPE_DEFAULT)),
-          replication, ozoneConfiguration);
+      ReplicationType replicationType = ReplicationType.valueOf(
+          ozoneConfiguration.get(OZONE_REPLICATION_TYPE, OZONE_REPLICATION_TYPE_DEFAULT));
+      clientConfiguredReplicationConfig =
+          (replicationType == ReplicationType.EC) ?
+              new ECReplicationConfig(replication) : ReplicationConfig.parse(
+              replicationType, replication, ozoneConfiguration);
     }
```

Comment:
I think this can be simplified to:

```java
ReplicationConfig clientConfiguredReplicationConfig = OzoneClientUtils.getClientConfiguredReplicationConfig(ozoneConfiguration);
```